msbuild/Xamarin.MacDev.Tasks/Tasks/WriteItemsToFile.cs

```diff
@@ -46,13 +43,14 @@ public override bool Execute ()
 					new XElement (ItemGroupElementName,
 						items.Select (item => this.CreateElementFromItem (item)))));
 
-			if (this.Overwrite && System.IO.File.Exists (this.File.ItemSpec))
-				System.IO.File.Delete (this.File.ItemSpec);
+			var file = this.File?.ItemSpec;
+			if (this.Overwrite && System.IO.File.Exists (file))
+				System.IO.File.Delete (file);
 
-			if (!Directory.Exists (Path.GetDirectoryName (this.File.ItemSpec)))
-				Directory.CreateDirectory (Path.GetDirectoryName (this.File.ItemSpec));
+			if (!Directory.Exists (Path.GetDirectoryName (file)))
+				Directory.CreateDirectory (Path.GetDirectoryName (file));
 
-			document.Save (this.File.ItemSpec);
+			document.Save (file);
```

Comment:
If I am not mistaken, file here can be null as per:
```csharp
var file = this.File?.ItemSpec;
```
and at least in dotnet7 if you pass null to save you'll get the following:
```csharp
Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'outputFileName')
   at System.ArgumentNullException.Throw(String paramName)
   at System.ArgumentNullException.ThrowIfNull(Object argument, String paramName)
   at System.Xml.XmlWriterSettings.CreateWriter(String outputFileName)
   at System.Xml.Linq.XDocument.Save(String fileName, SaveOptions options)

```

Reply:
this.File will never be null (it's a [Required] property, so MSBuild will show an error before we get to this point).

I could change it to `var file = this.File!.ItemSpec;`, although the final IL is the same.